pkg/k8s/endpointsynchronizer/cep.go

```diff
@@ -225,48 +227,72 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 					return nil
 				}
 
-				// We have no localCEP copy. We need to fetch it for updates, below.
-				// This is unexpected as there should be only 1 writer per CEP, this
-				// controller, and the localCEP created on startup will be used.
-				if localCEP == nil {
-					localCEP, err = ciliumClient.CiliumEndpoints(namespace).Get(podName, meta_v1.GetOptions{})
-					switch {
-					// The CEP doesn't exist in k8s. This is unexpetected but may occur
-					// if the endpoint was removed from k8s but not yet within the agent.
-					// Mark the CEP for creation on the next controller iteration. This
-					// may never occur if the controller is stopped on Endpoint delete.
-					case err != nil && k8serrors.IsNotFound(err):
-						needInit = true
+				switch {
+				case k8s.JSONPatchVerConstr.Check(k8sServerVer):
+					// For json patch we don't need to perform a GET for endpoints
+
+					// If it fails it means the test from the previous patch failed
+					// so we can safely replace this node in the CNP status.
+					replaceCEPStatus := []k8s.JSONPatch{
+						{
+							OP:    "replace",
+							Path:  "/status",
+							Value: mdl,
+						},
+					}
+					var createStatusPatch []byte
+					createStatusPatch, err = json.Marshal(replaceCEPStatus)
+					if err != nil {
```

Comment:
Should/Could this fallback onto using Update? I'm not sure that it's worth the added code complexity.

Reply:
I would prefer not too, it should work for k8s 1.13, if it fails we need to know it is not working